core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1520,46 +1551,106 @@ public long loadMetadata(AlluxioURI path, boolean recursive)
             ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(path.getPath()));
       }
       if (ufs.isFile(ufsUri.toString())) {
-        long ufsBlockSizeByte = ufs.getBlockSizeByte(ufsUri.toString());
-        long ufsLength = ufs.getFileSize(ufsUri.toString());
-        // Metadata loaded from UFS has no TTL set.
-        CreateFileOptions options =
-            CreateFileOptions.defaults().setBlockSizeBytes(ufsBlockSizeByte).setRecursive(recursive)
-                .setMetadataLoad(true).setPersisted(true);
-        long fileId = createFile(path, options);
-        CompleteFileOptions completeOptions =
-            CompleteFileOptions.defaults().setUfsLength(ufsLength);
-        completeFile(path, completeOptions);
-        return fileId;
+        synchronized (mInodeTree) {
```

Comment:
I did not want to wrap isFile inside the lock. I guess that is not a big deal for now since the mInodeTree lock is so big. Done